crates/bevy_reflect/derive/src/string_expression.rs

```diff
@@ -0,0 +1,106 @@
+use proc_macro2::TokenStream;
+use quote::{quote, ToTokens};
+use syn::{spanned::Spanned, LitStr};
+
+/// Contains tokens representing different kinds of string.
+#[derive(Clone)]
+pub(crate) enum StringExpression {
```

Comment:
Nit: It's not a huge deal, but I think I personally prefer `StringExpr`. In the context of proc-macros, it carries a bit more immediate understanding as it is reminiscent of `Expr` and its variants. In other words, you know immediately that it's meant for generating token streams as opposed to a general-purpose utility struct for strings (though, tbf, this rename isn't that far off either).

Reply:
I can rename it if you like, although abbreviations do sometimes bother me. For example, `Ref` and `Res` would be easier to read for me if they were called "Reference" and "Resource" (came across those while reading the conversation on a PR and didn't have the code or docs handy). I feel like if the max_line_width were higher, ppl wouldn't need to sv chars.

Reply:
Abbreviations are also harder to read for ESL